apps/mobile/src/navigation/MainTabNavigator/MainTabNavigator.tsx

```diff
@@ -11,14 +11,18 @@ import { HomeScreen } from '../../screens/HomeScreen/HomeScreen';
 
 const Tab = createMaterialTopTabNavigator<MainTabNavigatorParamList>();
 
-export function MainTabNavigator() {
+type MainTabNavigatorProps = {
+  hasUnreadNotifications: boolean;
+};
+
+export function MainTabNavigator({ hasUnreadNotifications }: MainTabNavigatorProps) {
   const { top } = useSafeAreaInsets();
 
   return (
     <Tab.Navigator
       tabBarPosition="bottom"
       initialRouteName="Home"
-      tabBar={TabBar}
+      tabBar={(props) => <TabBar {...props} hasUnreadNotifications={hasUnreadNotifications} />}
```

Comment:
We should try to avoid this pattern https://reactnavigation.org/docs/troubleshooting/#screens-are-unmountingremounting-during-navigation

This is quite annoying since the approach w/ Relay is to pass fragment refs down as props and we don't have a way to pass props to the component. I think the right thing to do here is to have the leaf component (NotificationTabItem) use a lazy loaded query.

While the query is loading, we can show a notification icon w/o the blue dot, and then when it loads, the user may or may not see the blue dot based on whether or not they have unread notifications.

Reply:
I pushed up a change for this since I had a very specific idea in mind